### PR TITLE
Mobile/coursepage

### LINKED ISSUE #403: [Mobile] Add Resource
### Issue

There needs to be a future to add resources to a topic. 

### Task(s)

- [x] Creating an add resource button.
- [x] Creating an add resource page
- [x] Connecting to the backend

### Deliverable(s)

* Add resource button and page.

### Acceptance Criteria

* Ability to add resources to a topic.

### Deadline of the issue

03.12.2022 @23.59

### Reviewer

None

### Deadline for Review

_No response_

### LINKED ISSUE #425: [Mobile] Delete Resource
### Issue

There needs to be a future to delete resources.

### Task(s)

- [x]  Creating a delete resource button.
- [x]  Connecting to the backend

### Deliverable(s)

* Delete resource option

### Acceptance Criteria

* Be able to delete resources

### Deadline of the issue

05.12.2022 @23.59

### Reviewer

None

### Deadline for Review

_No response_

### LINKED ISSUE #426: [Mobile] Edit Resource
### Issue

There needs to be a future to edit resources.

### Task(s)

- [x]  Creating an edit resource button.
- [x]  Creating an edit resource page
- [x]  Connecting to the backend 

### Deliverable(s)

* Edit button and edit page

### Acceptance Criteria

* Being able to edit resources

### Deadline of the issue

05.12.2022 @23.59

### Reviewer

None

### Deadline for Review

_No response_

### PR DESCRIPTION
**Pull request by salimtirit**
### Closes #403
- Created a topic page 
- Created add resource page #403 
  - Used markdown editor developed in #375
- Created edit resource page
  - Used markdown editor developed in #375
- Added delete resource option #425 
- Created add topic page #402

### CODE REVIEW

File: app/mobile/lib/view/resource/edit_resource_page.dart [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+                      Resource? created = await contentService
+                          .editResource(viewModel.controller_title.text, viewModel.controller.text, resourceId: r.id);
+                      if (created == null) {
+                        ScaffoldMessenger.of(context).showSnackBar(
+                          const SnackBar(
+                              content: Text('Could Not Edit Resource!')),
+                        );
+                        viewModel.notifyListeners();
+                      }
+                      if (created!.id.isNotEmpty) {
```
Comment (hsuleerkul) [inline app/mobile/lib/view/resource/edit_resource_page.dart:97]: This if statement is not correct. When the created = null, checking content as nonnullable creates an error.  I think changing it with following would be enough for solving it. 

```
if (created != null) {
    if (created!.id.isNotEmpty) {
        ScaffoldMessenger.of(context).showSnackBar(
            const SnackBar(content: Text('Successfully Editted!')),);
        Navigator.pop(context);
        viewModel.notifyListeners();
   }
} else {
    ScaffoldMessenger.of(context).showSnackBar(
         const SnackBar( content: Text('Could Not Edit Resource!')),);
         viewModel.notifyListeners();
 }
 ```

File: app/mobile/lib/view/topic/long_press_dialog.dart [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+                    color: CustomColors.main,
+                    borderRadius: BorderRadius.circular(10)),
+              ),
+            ),
+            GestureDetector(
+              onTap: () {
+                showDialog(
+                  context: context,
+                  builder: ((context) => AlertDialog(
+                        content: Column(
```
Comment (hsuleerkul) [inline app/mobile/lib/view/topic/long_press_dialog.dart:84]: I have experienced overflow in my device. Here is a screenshot:

File: app/mobile/lib/view/topic/long_press_dialog.dart [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+                                  onTap: () async {
+                                    int? deleted = await contentService
+                                        .deleteResource(resourceId: r.id);
+                                    if (deleted == 201) {
+                                      //TODO: remove the resource tile from here
+                                      ScaffoldMessenger.of(context)
+                                          .showSnackBar(
+                                        const SnackBar(
+                                            content: Text(
+                                                'Resource Successfully Deleted!')),
```
Comment (hsuleerkul) [inline app/mobile/lib/view/topic/long_press_dialog.dart:110]: Page does not refresh immidiately after deleting a resource. Here is a screenshot after I deleted resource 'How black is your Black' and at the end of the page it says 'Resource Successfully Deleted!', however, it is still visible on the page.

**Review by hsuleerkul — CHANGES_REQUESTED** [review]
I have specified the changes needed. ty

File: app/mobile/lib/view/resource/edit_resource_page.dart (hunk 1 above)
Comment (hsuleerkul) [reply app/mobile/lib/view/resource/edit_resource_page.dart]: I committed the changes in here: [843d3db](https://github.com/bounswe/bounswe2022group3/commit/843d3db978706dec3e9e230e10a7feeafcb97dba)

Comment (salimtirit) [reply app/mobile/lib/view/resource/edit_resource_page.dart]: Thank you for fixing my error :) I guess I was very distracted while writing this.

File: app/mobile/lib/view/topic/long_press_dialog.dart (hunk 2 above)
Comment (salimtirit) [reply app/mobile/lib/view/topic/long_press_dialog.dart:84]: I fixed this problem by changing the row widget to the wrap widget. It should be fine now.

File: app/mobile/lib/view/topic/long_press_dialog.dart (hunk 3 above)
Comment (salimtirit) [reply app/mobile/lib/view/topic/long_press_dialog.dart:110]: I fixed this problem with my last commit

File: app/mobile/lib/view/topic/long_press_dialog.dart (hunk 2 above)
Comment (hsuleerkul) [reply app/mobile/lib/view/topic/long_press_dialog.dart:85]: I have confirmed the new warning box. It's working :)

File: app/mobile/lib/view/topic/long_press_dialog.dart (hunk 3 above)
Comment (hsuleerkul) [reply app/mobile/lib/view/topic/long_press_dialog.dart:110]: This one also works.

**Comment by hsuleerkul** [comment]
I have reviewed the changes you made and everything looks good to me. Also bringing resource to the top of the page instead of the center is a nice touch.

**Pull request merged**